2.3-tapscript.ipynb

```diff
@@ -396,9 +403,9 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "#### 2.3.5 _Programming Exercise:_ Generate a 2-of-2 `csahasholder` tapscript\n",
+    "#### 2.3.5 _Programming Exercise:_ Generate a 2-of-2 `csa_hashlock_delay` tapscript\n",
     "\n",
-    "Construct a `csahasholder` tapscript with the following locking conditions:\n",
+    "Construct a `csa_hashlock_delay` tapscript with the following locking conditions:\n",
     "\n",
     "* 2-of-2 public keys\n",
     "* Hashlock with the preimage `sha256(b'secret')`\n",
```

Comment:
This is strange. You change the comment in the first commit and then revert it in a later commit.

Do you use the preimage `sha256(b'secret')` to enforce that the preimage provided in the witness is 32 bytes? If so, is that explained somewhere?

Reply:
> Do you use the preimage sha256(b'secret') to enforce that the preimage provided in the witness is 32 bytes? If so, is that explained somewhere?

Yes there's a 32B size guard in the output script - I will add explainer that the hashlock does this to protect against unfeasibly large pre-images at spending time (e.g. atomic swap between two chains with different data push size limits, where correlating pre-image size is only valid on one-chain).